rero_ils/modules/loans/utils.py

```diff
@@ -144,7 +145,43 @@ def is_item_available_for_checkout(item_pid):
 
 def can_be_requested(loan):
     """Check if record can be requested."""
-    if not loan.get('item_pid'):
+    if not loan.item_pid:
         raise Exception('Transaction on document is not implemented.')
     policy = get_circ_policy(loan)
     return policy.get('allow_requests')
+
+
+def loan_build_item_ref(loan_pid, loan):
+    """Build $ref for the Item attached to the Loan."""
+    base_url = current_app.config.get('RERO_ILS_APP_BASE_URL')
+    url_api = '{base_url}/api/{doc_type}/{pid}'
+    return {
+        '$ref': url_api.format(
+            base_url=base_url,
+            doc_type='items',
+            pid=loan.item_pid)
+    }
+
+
+def loan_build_patron_ref(loan_pid, loan):
+    """Build $ref for the Patron attached to the Loan."""
+    base_url = current_app.config.get('RERO_ILS_APP_BASE_URL')
+    url_api = '{base_url}/api/{doc_type}/{pid}'
+    return {
+        '$ref': url_api.format(
+            base_url=base_url,
+            doc_type='patrons',
+            pid=loan.patron_pid)
+    }
+
+
+def loan_build_document_ref(loan_pid, loan):
+    """Build $ref for the Document attached to the Loan."""
+    base_url = current_app.config.get('RERO_ILS_APP_BASE_URL')
+    url_api = '{base_url}/api/{doc_type}/{pid}'
+    return {
+        '$ref': url_api.format(
+            base_url=base_url,
+            doc_type='documents',
+            pid=loan.document_pid)
+    }
```

Comment:
I created a general method to do this in `modules/utils.py:get_ref_for_pid`. See usage here : https://github.com/rero/rero-ils/blob/dev/tests/ui/test_utils_app.py